test/OptionSettlement.t.sol

```diff
@@ -521,92 +524,105 @@ contract OptionSettlementTest is Test, NFTreceiver {
         assertEq(decodedOptionIdFromCTokenId2, oTokenId >> 96);
         assertEq(decodedClaimIndexFromCTokenId2, 2); // second claim
     }
 
-    function testFuzzGetDecodedIdComponents(uint256 optionId, uint256 claimId) public {
+    function testFuzzDecodeTokenId(uint256 optionId, uint256 claimId) public {
         optionId = bound(optionId, 0, type(uint160).max);
         claimId = bound(claimId, 0, type(uint96).max);
 
         uint256 testTokenId = claimId;
         testTokenId |= optionId << 96;
 
-        (uint160 decodedOptionId, uint96 decodedClaimId) = engine.getDecodedIdComponents(testTokenId);
+        (uint160 decodedOptionId, uint96 decodedClaimId) = engine.decodeTokenId(testTokenId);
         assertEq(decodedOptionId, optionId);
         assertEq(decodedClaimId, claimId);
     }
 
-    function testGetOptionFromEncodedId() public {
+    function testGetOptionForTokenId() public {
         IOptionSettlementEngine.Option memory option = IOptionSettlementEngine.Option({
             underlyingAsset: DAI_A,
             underlyingAmount: 1,
             exerciseAsset: USDC_A,
             exerciseAmount: 100,
-            exerciseTimestamp: uint40(block.timestamp),
+            earliestExerciseTimestamp: uint40(block.timestamp),
             expiryTimestamp: uint40(block.timestamp + 30 days),
             settlementSeed: 0,
-            nextClaimId: 0
+            nextClaimNum: 0
         });
-        uint256 oTokenId =
-            engine.newOptionType(DAI_A, uint40(block.timestamp), uint40(block.timestamp + 30 days), USDC_A, 1, 100);
+        uint256 optionId =
+            engine.newOptionType(DAI_A, 1, USDC_A, 100, uint40(block.timestamp), uint40(block.timestamp + 30 days));
 
         // Update struct values to match stored option data structure
-        bytes20 optionHash = bytes20(keccak256(abi.encode(option)));
-        uint160 optionKey = uint160(optionHash);
+        uint160 optionKey = uint160(bytes20(keccak256(abi.encode(option))));
 
         option.settlementSeed = optionKey; // settlement seed is initially equal to option key
-        option.nextClaimId = 1; // next claim ID has been incremented
+        option.nextClaimNum = 1; // next claim num has been incremented
 
-        assertEq(engine.getOptionFromEncodedId(oTokenId), option);
+        assertEq(engine.getOptionForTokenId(optionId), option);
+    }
+
+    function testFailGetClaimForTokenId() public {
+        uint256 optionId =
+            engine.newOptionType(DAI_A, 1, USDC_A, 100, uint40(block.timestamp), uint40(block.timestamp + 30 days));
+
+        vm.prank(ALICE);
+        uint256 claimId = engine.write(optionId, 7);
+
+        IOptionSettlementEngine.OptionLotClaim memory claim = engine.getClaimForTokenId(claimId);
+
+        emit log_named_uint("Amount written", claim.amountWritten);
+        // emit log_named_boolean("Claimed", claim.claimed);
```

Comment:
we can drop both logs here

Reply:
This was actually for my testing, it's covered in the 3rd TODO in Engine.sol